### PR TITLE
Minor improvements 

### PR DESCRIPTION
**Pull request by thalurur**
Signed-off-by: Ravi Thaluru <thalurur@users.noreply.github.com>

*Issue #, if available:*
N/A

*Description of changes:*
1. Porting additional check in rollover step that is missing  
2. Minor optimization, converting all retrieved matching indices to set for a more faster lookup when processing delete event in Coordinator
3. Updating schema version - to force all changes going forward to be applied everywhere

*CheckList:*
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by bowenlan-amzn — DISMISSED** [review]

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollover/AttemptRolloverStep.kt [hunk 1]
```diff
@@ -46,6 +46,14 @@ class AttemptRolloverStep(private val action: RolloverAction) : Step(name) {
...
         }
 
+        // If we have already rolled over this index then fail as we only allow an index to be rolled over once
+        if (context.metadata.rolledOver == true) {
+            logger.warn("$indexName was already rolled over, cannot execute rollover step")
+            stepStatus = StepStatus.FAILED
+            info = mapOf("message" to getFailedDuplicateRolloverMessage(indexName))
+            return this
+        }
+
```
Comment (dbbaughe) [inline src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollover/AttemptRolloverStep.kt:56]: I had removed this in this PR:
https://github.com/opensearch-project/index-management/commit/5ad90ffd14f6e3005d32cbd343d547e6da8e690b#diff-397f9b437e1345a5b77c0edca0ec48d77265ecb45eaaabbf868d1e9d6bb11116L75-L81

Comment (dbbaughe) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollover/AttemptRolloverStep.kt:56]: Are we missing this commit ^ now?

Comment (thalurur) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollover/AttemptRolloverStep.kt:56]: hmm.. I see, I will clean this up then

Comment (thalurur) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollover/AttemptRolloverStep.kt:56]: No, we have the new commit. I was just looking at the diff with internal repo and found this changed, wasn't clear if we missed porting this. I should have checked the history

**Review by dbbaughe — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Pull request merged**